snuba/migrations/runner.py

```diff
@@ -33,15 +38,19 @@
 
 
 def get_active_migration_groups() -> Sequence[MigrationGroup]:
-
-    return [
-        group
-        for group in MigrationGroup
-        if not (
-            group in OPTIONAL_GROUPS
-            and group.value in settings.SKIPPED_MIGRATION_GROUPS
-        )
-    ]
+    groups = []
+    for group in MigrationGroup:
+        if group.value in settings.READINESS_STATE_MIGRATION_GROUPS_ENABLED:
+            readiness_state = get_readiness_state(group)
+            if readiness_state.value in settings.SUPPORTED_STATES:
+                groups.append(group)
+        else:
+            if not (
+                group in OPTIONAL_GROUPS
+                and group.value in settings.SKIPPED_MIGRATION_GROUPS
+            ):
+                groups.append(group)
```

Comment:
if a migration is skipped it's active? I'm not sure I understand that logic

Reply:
okay I think my confusion here is that it's not clear what "active" means or there's a logic bug

> If it isn't then default to using the existing SKIPPED_MIGRATIONS_GROUPS to determine if it's active.

But shouldn't it be

```python3
            if not (
                group in OPTIONAL_GROUPS
                and group.value **not** in settings.SKIPPED_MIGRATION_GROUPS
            ):
                groups.append(group)
```

otherwise you're going to mark a migration that is skipped in this environment as active

Reply:
I misread the code and just didn't see the `not` in front of the condition 😅